scripts/ci/images/Dockerfile.ci-spack-ubuntu20.04-base

```diff
@@ -35,6 +35,19 @@ RUN . /spack/share/spack/setup-env.sh && \
       --fail-fast \
       --only dependencies \
       -j$(grep -c '^processor' /proc/cpuinfo) && \
+    spack env create adios2-ci-deps /tmp/specs.yaml && \
```

Comment:
It's fine if this is working, but I would have thought you'd want to process the dependencies environment first, then when you do the other environments, the `install` step for each one would make sure the dependency specs were installed in the environment as well.  So for each env, you'd add whatever adios2 spec, then add the dependency specs, then concretize, and finally install.  This way, you'd only have to `concretize` each adios2 environment once instead of twice.  Also, I don't think you'd need `concretize` or `install` steps for the dependencies environment in that case either, but maybe the `concretize` can't be avoided.